logging/taillogs/taillogs.go

```
@@ -0,0 +1,65 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taillogs
+
+// [START logging_tail_log_entries]
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+
+	logging "cloud.google.com/go/logging/apiv2"
+	loggingpb "google.golang.org/genproto/googleapis/logging/v2"
+)
+
+// tailLogs creates a channel to stream log entries that were recently ingested for a project
+func tailLogs(projectID string) {
+	// projectID := "your_project_id"
+
+	ctx := context.Background()
+	client, err := logging.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("NewClient error: %v", err)
```

log.Fatal is basically a panic. We should instead return an error from this function.

---

Replaced all `log.Fatalf` with `return fmt.Errorf`